fastlane/Fastfile

```diff
@@ -50,6 +54,35 @@ platform :ios do
     replace_ios_assets()
   end
 
+  lane :build_qa do
+    ipa_path = '../Mattermost.ipa'
+    File.delete(ipa_path) if File.file?(ipa_path)
+
+    match(type: 'adhoc', app_identifier: 'com.mattermost.rnbeta')
+    build_ios
+
+    plist_template = File.read('plist.erb')
+    abbreviated_commit_hash = `git rev-parse --short head`
+    abbreviated_commit_hash.strip!
+    # TODO: See if we can use abbreviated_commit_hash as bundle-version in the plist file
+    plist_body = ERB.new(plist_template).result(binding)
+
+    s3 = Aws::S3::Resource.new(region: ENV['AWS_REGION'])
+    plist_obj = s3.bucket(ENV['AWS_BUCKET_NAME']).object("ios/#{abbreviated_commit_hash}.plist")
+    plist_obj.put(body: plist_body)
+
+    ipa_obj = s3.bucket(ENV['AWS_BUCKET_NAME']).object("ios/#{abbreviated_commit_hash}.ipa")
+    ipa_obj.upload_file(ipa_path)
+
+    # TODO: Upload beta logo if not present
+
+    ios_install_url = "itms-services://?action=download-manifest&url=https://s3.#{ENV['AWS_REGION']}.amazonaws.com/#{ENV['AWS_BUCKET_NAME']}/ios/#{abbreviated_commit_hash}.plist"
+
+    if ENV['MATTERMOST_WEBHOOK_URL']
```

Comment:
I would try and use the `mattermost` plugin here for sending the post to MM, you can take a look at the `send_message_for_ios` and `send_message_for_android` functions that we it can be a little bit more awesome